Bluette1/stillspace-api
# Set Up For Vibes

## Pull Request: Set Up For Vibes

### Overview
This pull request introduces essential configurations to enhance the development workflow of the stillspace (Vibes) project. It adds GitHub Actions for continuous integration checks and sets up a precommit hook using Husky to ensure code quality.

### Changes Made
- **GitHub Actions**: Implemented CI checks to automate testing and ensure code integrity.
- **Precommit Hook**: 
  - Integrated RuboCop for Ruby code linting.
  - Configured Husky to enforce code quality checks before commits.

### Issues Addressed
This PR addresses issues #1 and #3 by streamlining the development process and improving code consistency.

### How to Test
1. Pull the branch `feature/add-still-space-api`.
2. Ensure that the GitHub Actions workflows trigger on push.
3. Test the precommit hook by attempting to commit code that does not meet RuboCop standards.

### Additional Notes
- All changes have been reviewed and tested locally.
- This setup will help maintain high code quality and prevent potential issues before merging into the main branch.

[User stories](https://www.notion.so/Still-Space-Vibes-API-11be6a4d98f280c0ab42d3407a0d117f?pvs=4)
